### ISSUE TITLE
Configure options to disable bin/doc/test

### ISSUE THREAD

**Issue by Kagami** [fribidi/fribidi#8]
Currently it's impossible to build fribidi without binary, documentation or tests. It cause some inconvenience because e.g. doc target requires `c2man` which is relative rare package and is not available on some systems.

I'm using

```
sed -i 's/^SUBDIRS =.*/SUBDIRS=gen.tab charset lib/' Makefile
```

hack to workaround this, but it would be nice to have options for this. Also, some distros e.g. Gentoo have [similar patches](https://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/dev-libs/fribidi/files/fribidi-0.19.2-nodoc.patch?revision=1.2&view=markup) for fribidi too.

Thanks.

**Comment by loqs** [comment]
If I understand the Makefile in the doc sub directory correctly it will do the preparation work for c2man if c2man is not found it will not generate usable results but will only display a warning.
Would adding configure test to check for the presence of c2man and either erroring if c2man is not found or not entering the doc dir if it is absent be acceptable?
Would adding --disable-doc etc to configure with make only entering the named subdirectory if the option is not passed be acceptable as well?

**Comment by behdad** [comment]
Or you can just do "make -C lib". What's wrong with that option?

**Comment by loqs** [comment]
That does not seem to work here
```
$ make -C lib
make: Entering directory '/tmp/fribidi/lib'
make[1]: Entering directory '/tmp/fribidi/gen.tab'
  CC       gen-unicode-version.o
  CCLD     gen-unicode-version
  GEN      fribidi-unicode-version.h
make[1]: Leaving directory '/tmp/fribidi/gen.tab'
make[1]: Entering directory '/tmp/fribidi/gen.tab'
  GEN      fribidi-unicode-version.h
  CC       gen-arabic-shaping-tab.o
  CCLD     gen-arabic-shaping-tab
  GEN      arabic-shaping.tab.i
make[1]: Leaving directory '/tmp/fribidi/gen.tab'
make[1]: Entering directory '/tmp/fribidi/gen.tab'
  CC       gen_bidi_type_tab-gen-bidi-type-tab.o
  CC       gen_bidi_type_tab-packtab.o
  CCLD     gen-bidi-type-tab
  GEN      bidi-type.tab.i
make[1]: Leaving directory '/tmp/fribidi/gen.tab'
make[1]: Entering directory '/tmp/fribidi/gen.tab'
  CC       gen_joining_type_tab-gen-joining-type-tab.o
  CC       gen_joining_type_tab-packtab.o
  CCLD     gen-joining-type-tab
  GEN      joining-type.tab.i
make[1]: Leaving directory '/tmp/fribidi/gen.tab'
make[1]: Entering directory '/tmp/fribidi/gen.tab'
  CC       gen_mirroring_tab-gen-mirroring-tab.o
  CC       gen_mirroring_tab-packtab.o
  CCLD     gen-mirroring-tab
  GEN      mirroring.tab.i
make[1]: Leaving directory '/tmp/fribidi/gen.tab'
make[1]: Entering directory '/tmp/fribidi/gen.tab'
  CC       gen_brackets_tab-gen-brackets-tab.o
  CC       gen_brackets_tab-packtab.o
  CCLD     gen-brackets-tab
  GEN      brackets.tab.i
Reading `unidata/BidiBrackets.txt'
make[1]: Leaving directory '/tmp/fribidi/gen.tab'
make[1]: Entering directory '/tmp/fribidi/gen.tab'
  CC       gen_brackets_type_tab-gen-brackets-type-tab.o
  CC       gen_brackets_type_tab-packtab.o
  CCLD     gen-brackets-type-tab
  GEN      brackets-type.tab.i
make[1]: Leaving directory '/tmp/fribidi/gen.tab'
git.mk: Generating .gitignore
make  all-am
make[1]: Entering directory '/tmp/fribidi/lib'
  CC       fribidi.lo
  CC       fribidi-arabic.lo
  CC       fribidi-bidi.lo
  CC       fribidi-bidi-types.lo
  CC       fribidi-deprecated.lo
  CC       fribidi-joining.lo
  CC       fribidi-joining-types.lo
  CC       fribidi-mirroring.lo
  CC       fribidi-brackets.lo
  CC       fribidi-run.lo
  CC       fribidi-shape.lo
cd ../lib && make  libfribidi-char-sets.la
make[2]: Entering directory '/tmp/fribidi/lib'
make[2]: *** No rule to make target 'libfribidi-char-sets.la'.  Stop.
make[2]: Leaving directory '/tmp/fribidi/lib'
make[1]: *** [Makefile:813: ../charset/libfribidi-char-sets.la] Error 2
make[1]: Leaving directory '/tmp/fribidi/lib'
make: *** [Makefile:432: all] Error 2
make: Leaving directory '/tmp/fribidi/lib'
```

**Comment by khaledhosny** [comment]
It should work after #59 is merged.

**Comment by loqs** [comment]
./configure --disable-bin --disable-test --disable-doc seems more intuitive to me than make -C lib
the result however is the same if you only wanted to disable one of bin or test that would be a different issue or would such a build fail anyway?

**Comment by behdad** [comment]
This is such an arbitrary request. Sure, docs needs extra things so makes sense to disable. But test / bin, I don't see what's the harm in building them.

**Comment by loqs** [comment]
Agreed it does seem arbitrary so I can understand you not wanting to implement it.
I can not speak for why Kagami wanted it.  If you would reject a patch implementing it on principal I can understand that as well in that case this issue should probably be closed.

**Comment by behdad** [comment]
I'm not making that call, just adding my $0.02. :)

**Comment by khaledhosny** [comment]
I generally agree with Behdad. Also lots of build options tend to bitrot and even conflict with each other over the time, so the less different build configurations to have the better.

**Comment by khaledhosny** [comment]
`make -C lib` works now and there is point and adding a configure option for every bit of the build system.
